src/index.ts

```diff
@@ -1,27 +1,55 @@
 import { ref, type Ref } from "@vue/reactivity"
-import { cloneDeep } from "lodash-es"
-import type { FormRoot, InputControl } from "./types"
+import { cloneDeep, groupBy } from "lodash-es"
+import type { FormErrorsState, FormRoot, UseFormControlOptions } from "./types"
 import type { PartialOrPrimitive } from "./types/utils"
 import { createControlsTree } from "./controlsTree"
+import { standardValidate } from "./validation"
+import type { InputControl } from "./types/controls"
 
-export const useFormControl = <TState, TValidatedState = TState>(
-  defaultState?: PartialOrPrimitive<TState>
-): FormRoot<TState> => {
-  type WrappedState = { inner: TState }
+export const useForm = <TState, TValidatedState = TState>(
+  defaultState?: PartialOrPrimitive<TState>,
+  options: UseFormControlOptions<TState, TValidatedState> = {}
+): FormRoot<TState, TValidatedState> => {
+  const { validationSchema } = options
 
-  const defaultFormState = ref({
-    inner: cloneDeep(defaultState)
-  }) as Ref<WrappedState>
-  const state = ref({ inner: cloneDeep(defaultState) }) as Ref<WrappedState>
+  const defaultFormState = ref(cloneDeep(defaultState)) as Ref<
+    TState | undefined
+  >
+  const state = ref(cloneDeep(defaultState)) as Ref<TState>
   const controlsCache = new Map<string, InputControl<unknown>>()
+  const errors = ref<FormErrorsState>({})
 
-  const controlsTree = createControlsTree<WrappedState>(
+  const controlsTree = createControlsTree<TState>(
     state,
     defaultFormState,
+    errors,
     controlsCache
   )
 
+  const validate = async () => {
```

Comment:
In my opinion, the best API for this would be something like this

```ts
const submit = async (onValidation: (value: TValidatedState) => void, onError?: (errors: ??) => void) => {}
```

So that you are not forced everythime to check the result. What do you think?